test/ice_nucleation_calibration.jl

```diff
@@ -0,0 +1,278 @@
+        C_l = FT(qₗ / ((1 - qₗ) * ϵₘ + qₗ))  # concentration/mol fraction of liquid
+        C_v = FT(10.8509 * 1e-6 - C_l)     # concentration/mol fraction of vapor
+        qᵥ = ϵₘ / (ϵₘ - 1 + 1 / C_v)
+        qᵢ = FT(0)
+
+        q = TD.PhasePartition.(qᵥ + qₗ + qᵢ, qₗ, qᵢ)
+        Rₐ = TD.gas_constant_air(tps, q)
+        eₛ = TD.saturation_vapor_pressure(tps, T₀, TD.Liquid())
+        e = eᵥ(qᵥ, p₀, Rₐ, R_v)
+        Sₗ = FT(e / eₛ)
+        IC = [Sₗ, p₀, T₀, qᵥ, qₗ, qᵢ, Nₐ, Nₗ, Nᵢ, FT(0)]
+        Sᵢ = ξ(tps, T₀) * Sₗ
+
+        heterogeneous = "ABIFM"
+        condensation_growth = "Condensation"
+        deposition_growth = "Deposition"
+        size_distribution = "Monodisperse"
+
+        params = (;
+            const_dt,
+            w,
+            heterogeneous,
+            condensation_growth,
+            deposition_growth,
+            size_distribution,
+        )
+
+        coeff_true = [FT(54.58834), FT(-10.54758)]
+    end
+
+    n_samples = 10
+    G_truth = run_model(params, coeff_true, IN_mode, FT, IC)     # ICNC from running parcel w default values
+    y_truth = zeros(length(G_truth), n_samples) # where noisy ICNC will be stored
+
+    dim_output = length(G_truth)
+    Γ = 0.01 * LinearAlgebra.I * (maximum(G_truth) - minimum(G_truth))
+    noise_dist = Distributions.MvNormal(zeros(dim_output), Γ)
+    y_truth = G_truth .+ rand(noise_dist)
+
+    # Generate initial ensemble and set up EKI
+    N_ensemble = 10     # runs N_ensemble trials per iteration
+    N_iterations = 15    # number of iterations the inverse problem goes through
+    initial_ensemble = EKP.construct_initial_ensemble(rng, prior, N_ensemble)
+    EKI_obj = EKP.EnsembleKalmanProcess(
+        initial_ensemble,
+        y_truth,
+        Γ,
+        EKP.Inversion();
+        rng = rng,
+    )
+
+    # Carry out the EKI calibration
+    # ϕ_n_values[iteration] stores ensembles of calibrated coeffs in that iteration
+    global ϕ_n_values = []
+    for n in 1:N_iterations
+        ϕ_n = EKP.get_ϕ_final(prior, EKI_obj)
+        G_ens = hcat(
+            [
+                run_model(params, ϕ_n[:, i], IN_mode, FT, IC) for
+                i in 1:N_ensemble
+            ]...,
+        )
+        EKP.update_ensemble!(EKI_obj, G_ens)
+
+        ϕ_n_values = vcat(ϕ_n_values, [ϕ_n])
+    end
+
+    # Mean coefficients of all ensembles in the final iteration
+    m_coeff_ekp = round(
+        Distributions.mean(ϕ_n_values[N_iterations][1, 1:N_ensemble]),
+        digits = 6,
+    )
+    c_coeff_ekp = round(
+        Distributions.mean(ϕ_n_values[N_iterations][2, 1:N_ensemble]),
+        digits = 6,
+    )
+
+    return [m_coeff_ekp, c_coeff_ekp, coeff_true[1], coeff_true[2]]
+end
+
+function test_J_calibration(FT, IN_mode)
+    output = calibrate_J_parameters(FT, IN_mode)
+    calibrated_parameters = [output[1], output[2]]
+    coeff_true = [output[3], output[4]]
+
+    TT.@test calibrated_parameters[1] ≈ coeff_true[1] rtol = FT(3.0)
```

Comment:
I think it's more related to the noise I added to the model output (ICNC) is our pseudo data. I could try using smaller noise or making the relative tolerance a percent of the true coefficients. It's big because the order of magnitude for the m coefficient is around 50.

Reply:
Relative tolerance is by definition the percentage?

Reply:
You're so right, my bad. I just fixed it to 15% and adjusted the prior for ABIFM so that it would fall in that range. Thanks Anna!